### PR TITLE
no-issue: Migrate to the pyproject.toml

### PR DESCRIPTION
**Pull request by corona10**

### CODE REVIEW

**Comment by corona10** [comment]
The next works will be adding the auto-release pipeline like pyperf: https://github.com/psf/pyperf/blob/main/.github/workflows/publish.yml

Can I get access to PyPI owner page?

**Comment by corona10** [comment]
'test / ubuntu-latest - 3.12-dev (pull_request)' is the expected failure.

**Comment by ericsnowcurrently** [comment]
> The next works will be adding the auto-release pipeline like pyperf:

a big +1

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,61 @@
+#  - git pull --rebase
+#  - Remove untracked files/dirs: git clean -fdx
+#  - maybe update version in pyperformance/__init__.py and doc/conf.py
+#  - set release date in doc/changelog.rst
+#  - git commit -a -m "prepare release x.y"
+#  - run tests: tox --parallel auto
+#  - git push
+#  - check the CI status:
+#    https://github.com/python/pyperformance/actions
```
Comment (ericsnowcurrently) [inline pyproject.toml:9]:
```suggestion
# Update dependencies:
#
#  - python3 -m pip install --user --upgrade pip-tools
#  - git clean -fdx  # remove all untracked files!
#  - (pip-compile --upgrade -o requirements.txt requirements.in)
#
# Prepare a release:
#
#  - git pull --rebase
#  - Remove untracked files/dirs: git clean -fdx
#  - maybe update version in pyperformance/__init__.py and doc/conf.py
#  - set release date in doc/changelog.rst
#  - git commit -a -m "prepare release x.y"
#  - run tests: tox --parallel auto
#  - git push
#  - check the CI status:
#    https://github.com/python/pyperformance/actions
```

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+classifiers = [
+    'Development Status :: 5 - Production/Stable',
+    'Intended Audience :: Developers',
+    'License :: OSI Approved :: MIT License',
+    'Natural Language :: English',
+    'Operating System :: OS Independent',
+    'Programming Language :: Python :: 3',
+    'Programming Language :: Python',
+]
+requires-python = ">=3.7"
```
Comment (ericsnowcurrently) [inline pyproject.toml:46]: Sadly, we've been using pyperformance with 3.6...

File: setup.py [hunk 3]
```diff
@@ -1,85 +0,0 @@
...
-        'name': 'pyperformance',
-        'version': VERSION,
-        'author': 'Collin Winter and Jeffrey Yasskin',
-        'license': 'MIT license',
-        'description': DESCRIPTION,
-        'long_description': long_description,
-        'url': 'https://github.com/python/benchmarks',
-        'classifiers': CLASSIFIERS,
-        'packages': find_packages(),
-        'include_package_data': True,
```
Comment (ericsnowcurrently) [inline setup.py:75]: What handles the package data in pyproject.toml?

File: pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+#  - git push
+#  - check the CI status:
+#    https://github.com/python/pyperformance/actions
+#
+# Release a new version:
+#
+#  - git tag VERSION
+#  - git push --tags
+#  - Remove untracked files/dirs: git clean -fdx
+#  - python -m build
```
Comment (ericsnowcurrently) [inline pyproject.toml:16]: This one line is the main difference in the workflow, right?

Where does the "build" module come from?  I don't have one installed.

**Review by ericsnowcurrently — COMMENTED** [review]
I've left a few comments.

Also, presumably you verified that the distribution package generated via pyproject.toml is identical to the one generated via setup.py.

File: pyproject.toml (hunk 2 above)
Comment (corona10) [reply pyproject.toml:46]: https://github.com/psf/pyperf/pull/142 
pyperf does not support Python3.6 now, but I am taking a look at the support that is available from now on.
But do we need to support EOL version at pyperformance by the way?

File: setup.py (hunk 3 above)
Comment (corona10) [reply setup.py:75]: https://github.com/python/pyperformance/pull/249/commits/f20a4635ee1798a8b6546af415ca8dfdfb078549

Updated! Thanks for catching!

File: pyproject.toml (hunk 4 above)
Comment (corona10) [reply pyproject.toml:24]: https://github.com/python/pyperformance/pull/249/commits/e5d73924218ffc32bf2ee35513f0e03d3f5d45df

Updated!

File: pyproject.toml (hunk 2 above)
Comment (corona10) [reply pyproject.toml:54]: FYI setuptools require Python3.7 from https://setuptools.pypa.io/en/latest/history.html#v59-7-0

**Review by ericsnowcurrently — APPROVED** [review]
LGTM

**Pull request merged**